Gems/RecastNavigation/Code/Tests/NavigationMeshTest.cpp

```diff
@@ -0,0 +1,637 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <MockInterfaces.h>
+#include <RecastNavigationSystemComponent.h>
+#include <AzCore/Component/ComponentApplication.h>
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Console/Console.h>
+#include <AzCore/EBus/EventSchedulerSystemComponent.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzCore/UnitTest/Mocks/MockITime.h>
+#include <AzFramework/Entity/EntityDebugDisplayBus.h>
+#include <AzFramework/Physics/PhysicsScene.h>
+#include <Components/DetourNavigationComponent.h>
+#include <Components/RecastNavigationMeshComponent.h>
+#include <Components/RecastNavigationPhysXProviderComponent.h>
+
+namespace RecastNavigationTests
+{
+    using namespace AZ;
+    using namespace AZStd;
+    using namespace RecastNavigation;
+    using namespace testing;
```

Comment:
update: Please remove these using directives and instead use using declarations where appropriate. (can share info on why they're bad offline 😛

Reply:
Removed, done.

Reply:
Amazing thank you! 😄